### DISCUSSION TITLE
How to use team members?

### DISCUSSION THREAD

**Discussion by theronic** [authzed/spicedb#1863] [Q&A] [answered]
I'm trying to understand how to use team membership but struggling to get the relations right (schema membership examples don't seem to illustrate how to assign relations so members inherit team permissions).

I am trying to assign users to a team so that all members of a particular team has the `reboot` permission on `resource`(s) falling under a particular account. Given the following schema,

```zed
definition user {}

definition team {
  member: user
}

definition account {
  relation owner: user
  permission admin = owner
}

definition resource {
  relation owner: user | account | team#member
  permission reboot: owner + owner->admin
}
```

My understanding is that I should be able to create a team "team:developers-acme", add my-user to the team, and...somehow give all team members the reboot permission for resources falling under account `account:acme`.

What would the write permissions look like to achieve this? Or is this not possible?

**Comment by vroldanbet** [comment, marked as answer]
@theronic [I created this playground example](https://play.authzed.com/s/ZkocFxbmP51V/relationships) with your schema, illustrating how it works. Does it help?

The relationships to write are:
```
resource:prod#owner@team:sre#member
team:sre#member@user:jane
team:sre#member@user:joe
```

**Reply by theronic** [reply]
Thanks, @vroldanbet! In my Clojure implementation, I initially missed the `optional_relation` field in SubjectReference, thinking it was a stand-in for the relationship relation, so I was unable to write `team:sre#member`.
